### PR TITLE
parser: recognise \r\n as whitespace

### PR DESCRIPTION
**Pull request by tiancaiamao**

### CODE REVIEW

File: parser/lexer.go [hunk 1]
```diff
@@ -125,7 +125,7 @@ func (s *Scanner) skipWhitespace() rune {
 
 func (s *Scanner) scan() (tok int, pos Pos, lit string) {
 	ch0 := s.r.peek()
-	if isWhitespace(ch0) {
+	if unicode.IsSpace(ch0) {
```
Comment (shenli) [inline parser/lexer.go]: How about there is '\r\n' in a blob data? And please add a few test cases for this.

Comment (tiancaiamao) [inline parser/lexer.go]: https://dev.mysql.com/doc/refman/5.7/en/identifiers.html

unquoted identifiers can't contain \r \n
if they appears in other place, quoted identifiers begin with ` , and strings begin with ' or ", all have special opening character.

I'll add some tests tomorrow.

**Comment by shenli** [comment]
LGTM
@coocood @zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Comment by shenli** [comment]
@tiancaiamao Have you passed tidb-test?

**Comment by tiancaiamao** [comment]
yes, tidb-test passed.

**Pull request merged**